src/internal_events/datadog_traces.rs

```diff
@@ -1,21 +1,23 @@
+use crate::{emit, internal_events::ComponentEventsDropped};
 use metrics::counter;
 use vector_core::internal_event::InternalEvent;
 
 use super::prelude::{error_stage, error_type};
 
 #[derive(Debug)]
 pub struct DatadogTracesEncodingError {
-    pub message: &'static str,
+    pub error_message: &'static str,
+    pub error_reason: String,
     pub dropped_events: u64,
-    pub reason: String,
 }
 
 impl InternalEvent for DatadogTracesEncodingError {
     fn emit(self) {
+        let reason = "Failed to encode Datadog traces.";
         error!(
-            message = "Failed to encode Datadog traces.",
-            error = %self.message,
-            error_reason = %self.reason,
+            message = reason,
+            error = %self.error_message,
+            error_reason = %self.error_reason,
             error_type = error_type::ENCODER_FAILED,
             stage = error_stage::PROCESSING,
```

Comment:
@jszwedko we need to align on this tag - I don't believe we're using it consistently.

Reply:
Hmm, yeah, looks like `stage` and friends got dropped from the spec by https://github.com/vectordotdev/vector/commit/af1905a0648c56f0e82f315b4a60845ca4438616

I think that was accidental though. I'll revive them.

Reply:
To make sure I'm aligned- this error! log would not need to be modified?

Reply:
Yeah, I think the tags here are fine.

Reply:
https://github.com/vectordotdev/vector/pull/14153